confignode/src/main/java/org/apache/iotdb/confignode/consensus/response/SchemaPartitionDataSet.java

```diff
@@ -20,63 +20,75 @@
+
+    schemaPartition
+        .getSchemaPartitionMap()
         .forEach(
-            entity -> {
-              schemaRegionReplicaSets.putIfAbsent(entity.getKey(), new HashMap<>());
-              entity
-                  .getValue()
-                  .entrySet()
-                  .forEach(
-                      replica -> {
-                        RegionReplicaSet regionReplicaSet = new RegionReplicaSet();
-                        regionReplicaSet.setRegionId(replica.getValue().getId().getId());
-                        List<EndPoint> endPoints = new ArrayList<>();
-                        replica
-                            .getValue()
-                            .getDataNodeList()
-                            .forEach(
-                                dataNode -> {
-                                  EndPoint endPoint =
-                                      new EndPoint(
-                                          dataNode.getEndPoint().getIp(),
-                                          dataNode.getEndPoint().getPort());
-                                  endPoints.add(endPoint);
-                                });
-                        regionReplicaSet.setEndpoint(endPoints);
-                        schemaRegionReplicaSets
-                            .get(entity.getKey())
-                            .put(replica.getKey().getDeviceGroupId(), regionReplicaSet);
-                      });
+            (storageGroup, seriesPartitionSlotRegionReplicaSetMap) -> {
+              // Extract StorageGroupName
+              schemaRegionMap.putIfAbsent(storageGroup, new HashMap<>());
+
+              // Extract Map<SeriesPartitionSlot, RegionReplicaSet>
+              seriesPartitionSlotRegionReplicaSetMap.forEach(
+                  ((seriesPartitionSlot, regionReplicaSet) -> {
+                    TRegionReplicaSet regionMessage = new TRegionReplicaSet();
+                    regionMessage.setRegionId(regionReplicaSet.getId().getId());
+                    List<EndPoint> endPointList = new ArrayList<>();
+                    regionReplicaSet
+                        .getDataNodeList()
+                        .forEach(
+                            dataNodeLocation ->
+                                endPointList.add(
+                                    new EndPoint(
+                                        dataNodeLocation.getEndPoint().getIp(),
+                                        dataNodeLocation.getEndPoint().getPort())));
+                    regionMessage.setEndpoint(endPointList);
+                    schemaRegionMap
+                        .get(storageGroup)
+                        .put(seriesPartitionSlot.getDeviceGroupId(), regionMessage);
+                  }));
             });
-    rpcSchemaPartitionInfo.setSchemaRegionDataNodesMap(schemaRegionReplicaSets);
-    return rpcSchemaPartitionInfo;
+
+    // resp.setSchemaRegionMap(schemaRegionMap);
```

Comment:
Is this right?

Reply:
I'm not sure currently. I first update the process of get or create SchemaPartition in ConfigNode. But then the disscation on Friday decides to change interfaces about SchemaPartition. And the PatternTree isn't ready to use for ConfigNode now. So I comment this codes casually. I'll check this codes and support new SchemaPartition interfaces in next PR.